src/karthik/FindFinalDirectionOFVehicle/findFinalDirectionOfVehicle.spec.ts

```
@@ -0,0 +1,77 @@
+import { inputData } from './data';
+import {
+  findFinalDirectionOfVehicle,
+  validMoves
+} from './findFinalDirectionOfVehicle';
+
+describe('findFinalDirectionOfVehicle function', () => {
+  test('validity 1 for input data', () => {
+    expect(findFinalDirectionOfVehicle(inputData, 'NORTH')).toEqual('SOUTH');
+  });
+
+  test('validity 2 for expected directions', () => {
+    expect(findFinalDirectionOfVehicle(['STRAIGHT'], 'NORTH')).toEqual('NORTH');
+  });
+
+  test('validity 3 for expected directions', () => {
+    expect(findFinalDirectionOfVehicle(['RIGHT'], 'NORTH')).toEqual('EAST');
+  });
+
+  test('validity 4 for expected directions', () => {
+    expect(findFinalDirectionOfVehicle(['BACK'], 'NORTH')).toEqual('SOUTH');
+  });
+
+  test('validity 5 for expected directions', () => {
+    expect(findFinalDirectionOfVehicle(['LEFT'], 'NORTH')).toEqual('WEST');
+  });
+
+  test('validity 6 for turnaround move 1', () => {
+    expect(
+      findFinalDirectionOfVehicle(['RIGHT', 'RIGHT', 'RIGHT', 'RIGHT'], 'NORTH')
+    ).toEqual('NORTH');
+  });
+
+  test('validity 7 for turnaround move 2', () => {
+    expect(findFinalDirectionOfVehicle(['BACK', 'BACK'], 'NORTH')).toEqual(
+      'NORTH'
+    );
+  });
+
+  test('validity 8 for Empty moves', () => {
+    expect(findFinalDirectionOfVehicle([], 'NORTH')).toEqual('NORTH');
+  });
+
+  test('validity 9 for Invalid moves', () => {
+    expect(findFinalDirectionOfVehicle(['S'], 'NORTH')).toEqual(
```

Shall expect the function to throw error in case of invalid move instead of returning a string with the error content.

---

done.